### PR TITLE
Progress log table

### LINKED ISSUE #4605: Add a table to record the history of campaign quantity progress
`dosomething_reportback_progress_log` table to store campaign goal total hour by hour

on cron store the total quantity towards the goal 

| id | campaign_id | timestamp | quantity |
| --- | --- | --- | --- |
| 1 | 363 | 1435075581 | 24 |
| 2 | 363 | 1435076581 | 40 |

### PR DESCRIPTION
**Pull request by angaither**
Fixes #4605

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install [hunk 1]
```diff
@@ -228,6 +228,35 @@ function dosomething_reportback_schema() {
...
+        'not null' => TRUE,
+      ),
+      'timestamp' => array(
+        'description' => 'The Unix timestamp of the time this quantity was stored.',
+        'type' => 'int',
+        'not null' => TRUE,
+        'default' => 0,
+      ),
+    ),
+    'primary key' => array('id', 'nid'),
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install]: Why both `'id'` and `'nid'`?

Comment (angaither) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install]: so it's easier to query search, it looked like need both, when I was testing old records were getting overwritten on each cron

Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install]: They you should declare these as an index: https://api.drupal.org/api/drupal/includes!database!schema.inc/group/schemaapi, see `indexes`
Primary index is nor for search, but for unique identification of each row.

Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.install]: Auto-incremental `id` is sufficient for this purpose.

**Comment by sergiitk** [comment]
:+1:

**Pull request merged**